### ISSUE TITLE
Buttons use colour alone to convey state change: impacting WCAG 1.4.1

### ISSUE THREAD

**Issue by chris-gds** [tailwindlabs/tailwindcss#12054]
## Reproduction URL

https://tailwindui.com/components/application-ui/elements/buttons

![tailwind-buttons](https://github.com/tailwindlabs/tailwindcss/assets/71266765/f134f314-a7a8-4539-81e3-da8380fba994)

---

## Describe your issue

The hover states for buttons uses colour alone to convey a state change. [This impacts WCAG 1.4.1](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html)

### Further reading:
https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-without-color.html

https://govtnz.github.io/web-a11y-guidance/wct/buttons/make-a-button-accessible/buttons-colour-and-contrast.html#:~:text=A%20button%20must%20not%20rely,Knowledge%20Area%3A%20Colour%20and%20contrast.

- Users with partial sight often experience limited color vision.
- Some older users may not be able to see color well.
- Users who have color-blindness benefit when information conveyed by color is available in other visual ways.
- People using text-only, limited color, or monochrome displays may be unable to access color-dependent information.
- Users who have problems distinguishing between colors can look or listen for text cues.
- People using Braille displays or other tactile interfaces can detect text cues by touch.

## Solution

Add an additional visual change to states, border, underline etc.
